catalog/spatial/geocoding/spatial-geocoding-feature/src/main/java/org/codice/ddf/spatial/geocoding/feature/CatalogFeatureIndexer.java

```diff
@@ -0,0 +1,170 @@
+import org.codice.ddf.spatial.geocoding.GeoCodingConstants;
+import org.opengis.feature.simple.SimpleFeature;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CatalogFeatureIndexer implements FeatureIndexer {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(CatalogFeatureIndexer.class);
+
+  private static final ThreadLocal<WKTWriter> WKT_WRITER_THREAD_LOCAL =
+      ThreadLocal.withInitial(WKTWriter::new);
+
+  private Security security = Security.getInstance();
+
+  private CatalogFramework catalogFramework;
+
+  private CatalogHelper catalogHelper;
+
+  public void setSecurity(Security security) {
+    this.security = security;
+  }
+
+  public CatalogFeatureIndexer(CatalogFramework catalogFramework, CatalogHelper catalogHelper) {
+    this.catalogFramework = catalogFramework;
+    this.catalogHelper = catalogHelper;
+  }
+
+  @Override
+  public void updateIndex(
+      String resource, FeatureExtractor featureExtractor, boolean create, IndexCallback callback)
+      throws FeatureExtractionException, FeatureIndexingException {
+    Validate.notNull(featureExtractor, "featureExtractor can't be null");
+
+    if (create) {
+      removeExistingMetacards();
+    }
+
+    final AtomicInteger extractionCount = new AtomicInteger();
+
+    featureExtractor.pushFeaturesToExtractionCallback(
+        resource, feature -> handleFeatureExtraction(feature, create, callback, extractionCount));
+  }
+
+  private void handleFeatureExtraction(
+      SimpleFeature feature,
+      boolean create,
+      IndexCallback indexCallback,
+      AtomicInteger extractionCount)
+      throws FeatureIndexingException {
+    createOrUpdateMetacardForFeature(feature, create);
+    indexCallback.indexed(extractionCount.incrementAndGet());
+  }
+
+  private void createOrUpdateMetacardForFeature(SimpleFeature feature, boolean create)
+      throws FeatureIndexingException {
+    try {
+      security.runWithSubjectOrElevate(
+          () -> {
+            Metacard metacard = null;
+
+            if (!create) {
+              metacard = findMetacardForFeature(feature);
+            }
+            if (metacard == null) {
+              metacard = createMetacardForFeature(feature);
+              catalogFramework.create(new CreateRequestImpl(metacard));
+            } else {
+              catalogFramework.update(new UpdateRequestImpl(metacard.getId(), metacard));
+            }
+            return null;
+          });
+    } catch (SecurityServiceException | InvocationTargetException e) {
+      LOGGER.debug("Failed to index feature", e);
+      throw new FeatureIndexingException(e.getMessage());
+    }
+  }
+
+  private Metacard createMetacardForFeature(SimpleFeature feature) throws FeatureIndexingException {
+    Metacard metacard = new MetacardImpl();
+    String countryCode = feature.getID();
+    metacard.setAttribute(new AttributeImpl(Core.TITLE, countryCode));
```

Comment:
✏️ nit - if you are going to use MetacardImpl, you could just use the `setTitle` convenience method